### ISSUE TITLE
Manual command kbar toggle not working

### ISSUE THREAD

**Issue by jseanpatel** [jseanpatel/portfolio-v3#23]
**Describe the bug**

I'm trying to add functionality to manually toggle the command bar. The problem is that when I run `query.toggle` according to this documentation, nothing is happening.

**To Reproduce**
1. Clone this repository
2. Checkout the `kbar-error` branch
3. Click on the button all the way to the right in the `Navbar`

The component for toggling the command bar is contained in `CommandPaletteToggle`.

**Expected behavior**
We are expecting the prompt to show up just like it would if you used the key combo `cmd+k`

**Screenshots**
![Screen Shot 2022-08-01 at 10 39 57 PM](https://user-images.githubusercontent.com/31494454/182300047-cd781694-97a1-4b8e-9901-0920ba302409.png)
_Button is all the way on the right_

**Changes made**
- I added the `CommandPaletteToggle` component and its file entirely, the component can be found to the far right side of the Navbar
- I added the `CommandPaletteToggle` in `LayoutWrapper.js` which is how the component is actually displayed on the Navbar
- I added a wrapper `KBarProvider` in `app.js` as instructed so that I could import the `useKBar` hook in `CommandPaletteToggle`. 

**Additional context**
[Here](https://github.com/timc1/kbar/blob/16ae432abb832324387ad90ae23dcefbcb9c42e3/example/src/Layout.tsx#L15) is a working example that I based my attempt off of.

I found some commentary that the component where I am using the `useKBar` hook (on line 6 of `CommandPaletteToggle`) has to be wrapped with a `KBarProvider` component which I did on line 26 in `app`.

I've also tried `query.toggle()` but no luck.

**Comment by lucaspatel** [comment]
You did not check in any code that places a button on the right. @jseanpatel

**Comment by jseanpatel** [comment]
You can see the live version on the existing website [here](https://jseanpa.tel).

**Comment by lucaspatel** [comment]
The problem was that you had a nested `<KBarProvider>` (from both `_app.js` and within `CommandPalette.js`). I removed the one from `_app.js` and this fixed the double listener for cmd+k, but the button now broke the page since the `<KBarProvider>` was not reachable. To ameliorate this, I reworked `CommandPalette.js` to export its own named `<KBarProvider>` and now everything works. This solution is nice because it both fixes the problem and allows you to keep your command palette logic encapsulated into `CommandPalette.js`. Nicely done.

Fixed in #24, see the PR for more details.
